pkg/terra/client/gas_price_estimator.go

```diff
@@ -0,0 +1,128 @@
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type GasPriceEstimator interface {
+	MustGasPrice(denom []string) []sdk.DecCoin
+}
+
+var _ GasPriceEstimator = (*FixedGasPriceEstimator)(nil)
+
+type FixedGasPriceEstimator struct {
+	gasPrice sdk.DecCoin
+}
+
+func NewFixedGasPriceEstimator(price sdk.DecCoin) *FixedGasPriceEstimator {
+	return &FixedGasPriceEstimator{gasPrice: price}
+}
+
+func (gpe *FixedGasPriceEstimator) MustGasPrice(denoms []string) []sdk.DecCoin {
+	return []sdk.DecCoin{gpe.gasPrice}
+}
+
+type FCDGasPriceEstimator struct {
+	fcdURL url.URL
+	prices prices
+	client http.Client
+	lggr   Logger
+}
+
+func NewFCDGasPriceEstimator(fcdURLRaw string, requestTimeout time.Duration, lggr Logger) (*FCDGasPriceEstimator, error) {
+	// Sanity check the URL works and populate the cached value
+	fcdURL, err := url.Parse(fcdURLRaw)
+	if err != nil {
+		return nil, err
+	}
+	client := http.Client{Timeout: requestTimeout}
+	gpe := FCDGasPriceEstimator{fcdURL: *fcdURL, client: client, lggr: lggr}
+	initialGasPrice, err := gpe.request()
```

Comment:
I'm ok with having the node check the URL on boot and fail fast, but would still like this to be a separate explicit method instead of the constructor.

Reply:
I went back and forth on this, but the current approach is nice because it keeps the fixed vs fcd based gas pricers separate with the intention that a user could chain them together i.e. if we can't build an fcd one with a given fcd url, build a fixed price one etc. instead of having the fcd one also potentially be a fixed price one (via fallback). 

You could add a separate SetInitialGasPrices method, but then you'd be force to having a fixed price fallback to handle the case where SetInitialGasPrices isn't called.

Reply:
Found an even better way actually